# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: 09db50c8f370
# Full Hash: 09db50c8f370b69f186f3e6b7ce6154a984b40b4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Pass codec configs by value/reference instead of pointers. r=ng
#   
#   This patch mainly sets the stage for using codec configs with StateMirroring.
#   It also makes the code a bit simpler, and removes the need to handle the nullptr
#   case.
# ==============================================================================

diff -r d1552c620c11 -r 09db50c8f370 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Wed Aug 11 11:33:21 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Wed Apr 14 17:36:40 2021 +0200
@@ -127,7 +127,7 @@
    * restarting transmission sub-system on the engine.
    */
   MediaConduitErrorCode ConfigureSendMediaCodec(
-      const VideoCodecConfig* codecInfo,
+      const VideoCodecConfig& codecInfo,
       const RtpRtcpConfig& aRtpRtcpConfig) override;
 
   /**
@@ -141,7 +141,7 @@
    * restarting transmission sub-system on the engine.
    */
   MediaConduitErrorCode ConfigureRecvMediaCodecs(
-      const std::vector<UniquePtr<VideoCodecConfig>>& codecConfigList,
+      const std::vector<VideoCodecConfig>& codecConfigList,
       const RtpRtcpConfig& aRtpRtcpConfig) override;
 
   /**
@@ -349,11 +349,11 @@
       mEngineReceiving;  // if true ==> Receive Subsystem up and running
 
   // Local database of currently applied receive codecs. Call thread only.
-  nsTArray<UniquePtr<VideoCodecConfig>> mRecvCodecList;
+  std::vector<VideoCodecConfig> mRecvCodecList;
 
   // Written only on the Call thread. Guarded by mMutex, except for reads on the
   // Call thread.
-  UniquePtr<VideoCodecConfig> mCurSendCodecConfig;
+  Maybe<VideoCodecConfig> mCurSendCodecConfig;
 
   // Bookkeeping of stats for telemetry. Call thread only.
   RunningStat mSendFramerate;